src/sync.ts

```diff
@@ -48,16 +50,10 @@ export class Sync {
         startUpSetting.gist != null && startUpSetting.gist !== "";
 
       if (gistAvailable === true && startUpSetting.autoDownload === true) {
-        vscode.commands
-          .executeCommand("extension.downloadSettings")
-          .then(() => {
-            if (startUpSetting.autoUpload && tokenAvailable && gistAvailable) {
```

Comment:
Removing the then part , allows it work as propsosed by adding this line. Have you tested it?

The test case should be it should not hit `autoUpload` when downloading process is completed **through** `autoDownload` and `settings.json` is changed ?

Reply:
Yes, I've tested it and that solution is probably better, so I have reverted to using then.